### ISSUE TITLE
[10] base_multi_company _compute_company_id improvement suggestion

### ISSUE THREAD

**Issue by sharonshentc** [OCA/multi-company#85]
Hi, can you make some changes to _compute_company_id in base_multi_company? It can fix some potential issues... it will definitely fix the current issue we are having inheriting from this module.

Replace:
```python
@api.multi
    def _compute_company_id(self):
        for record in self:
            for company in record.company_ids:
                    if company.id in self.env.user.company_ids.ids:
                        record.company_id = company.id
                        break
```
With this:
```python
@api.multi
    def _compute_company_id(self):
        for record in self:
            if self.env.user.company_id.id in record.company_ids.ids:
                record.company_id = self.env.user.company_id
            else:
                for company in record.company_ids:
                    if company.id in self.env.user.company_ids.ids:
                        record.company_id = company.id
                        break
```
So now, it will take the logged user's current company instead of taking the first company allowed of that logged user. I left the original code in "else", in case it's a special user that can access that record even if his current company is not in the allowed company of that record. 

If this suggestion is no good, please let us know why to help us understand.
Thank you!

**Comment by pedrobaeza** [comment]
Please make a PR (https://odoo-community.org/page/code) rewriting it this way:
```python
@api.multi
def _compute_company_id(self):
    for record in self:
        if self.env.user.company_id in record.company_ids:
            record.company_id = self.env.user.company_id
        else:
            record.company_id = (
                self.env.user.company_ids - record.company_ids
            )[:1]
```

**Comment by sharonshentc** [comment]
Hi, I don't understand your code? what does (self.env.user.company_ids - record.company_ids) do? it intersects the 2 lists? (I think this is what we want, the intersection of the 2 lists...)

**Comment by pedrobaeza** [comment]
Set operation called difference

**Comment by pedrobaeza** [comment]
But you are right. It's intersection what is needed. Please change it to `self.env.user.company_ids & record.company_ids`

**Comment by sharonshentc** [comment]
i thought the minus was overwritten by odoo or something...

**Comment by pedrobaeza** [comment]
Well, indeed each operator can be overwritten in Python for adding your own logic about what the operation means for you, and it's done in that way with recordset, but as said, it's intersection what we want.

**Comment by sharonshentc** [comment]
yeah, i mean i thought the minus was overwritten in odoo to do intersection, which would be crazy, but since i'm new to python and odoo, i wasn't sure :)

**Comment by pedrobaeza** [comment]
Sorry, my bad

**Comment by sharonshentc** [comment]
no problem

**Comment by sharonshentc** [comment]
Sorry but i think i will use the original code... record.company_ids is actually res.company.assignment. The code you suggested needs to be converted to sets and it also give me apple-orange error...

**Comment by pedrobaeza** [comment]
Fixed after merging #91
